src/main/java/frc/robot/subsystems/intake/BallIntakeIO.kt

```diff
@@ -0,0 +1,88 @@
+package frc.robot.subsystems.intake
+
+import com.revrobotics.CANSparkMax
+import com.revrobotics.CANSparkMaxLowLevel
+import com.revrobotics.SparkMaxAbsoluteEncoder
+import edu.wpi.first.math.geometry.Rotation2d
+import edu.wpi.first.math.util.Units
+import org.littletonrobotics.junction.LogTable
+import org.littletonrobotics.junction.inputs.LoggableInputs
+
+interface IntakeIO  {
+
+    class Inputs : LoggableInputs {
+        var position = Rotation2d()
+        var armVelocity = Rotation2d()
+
+            override fun toLog(table: LogTable?) {
+                table?.put("Intake Arm Position", position.radians)
+                table?.put("Intake Arm Speed", armVelocity.radians)
+            }
+
+
+            override fun fromLog(table: LogTable?) {
+                table?.getDouble("Intake Arm Position", position.radians)?.let { position = Rotation2d(it) }
+                table?.getDouble("Intake Arm Speed", armVelocity.radians)?.let { armVelocity = Rotation2d(it) }
+            }
+    }
+
+    fun updateInputs(inputs: Inputs)
+
+
+    fun setArmSpeed(speed: Double)
```

Comment:
A lot of these io interfaces do not need all of these methods. we will never simply set the arm speed. if you want to keep these you should probably add brackets so the implementing classes are not forced to override functions that will never be called

Reply:
You implemented these...

Reply:
i didnt lmao

Reply:
Deferred until after the merge.